examples/tv-app/android/java/MessagesManager.cpp

```diff
@@ -0,0 +1,425 @@
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+
+        jint length = env->GetArrayLength(messagesList);
+
+        for (jint i = 0; i < length; i++)
+        {
+            jobject messageObject = env->GetObjectArrayElement(messagesList, i);
+            jclass messageClass   = env->GetObjectClass(messageObject);
+
+            jfieldID getMessageIdField = env->GetFieldID(messageClass, "messageId", "Ljava/lang/String;");
+            jstring jmessageId         = static_cast<jstring>(env->GetObjectField(messageObject, getMessageIdField));
+            JniUtfString messageId(env, jmessageId);
+            if (jmessageId != nullptr)
+            {
+                uint8_t buf[kMessageIdLength];
+                VerifyOrReturnValue(chip::Encoding::HexToBytes(messageId.charSpan().data(), messageId.charSpan().size(), buf,
+                                                               sizeof(buf)) == sizeof(buf),
+                                    CHIP_ERROR_INVALID_ARGUMENT, ChipLogError(Zcl, "HexToBytes failed"));
+
+                ReturnErrorOnFailure(encoder.Encode(ByteSpan(buf, sizeof(buf))));
+            }
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "MessagesManager::HandleGetMessages status error: %s", err.AsString());
+    }
+
+    return err;
+}
+
+CHIP_ERROR MessagesManager::HandlePresentMessagesRequest(
+    const ByteSpan & messageId, const MessagePriorityEnum & priority, const BitMask<MessageControlBitmap> & messageControl,
+    const DataModel::Nullable<uint32_t> & startTime, const DataModel::Nullable<uint16_t> & duration, const CharSpan & messageText,
+    const Optional<DataModel::DecodableList<MessageResponseOption>> & responses)
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrReturnError(env != nullptr, CHIP_JNI_ERROR_NULL_OBJECT, ChipLogError(Zcl, "Could not get JNIEnv for current thread"));
+    JniLocalReferenceScope scope(env);
+
+    ChipLogProgress(Zcl, "Received MessagesManager::HandlePresentMessagesRequest");
+    VerifyOrReturnError(mMessagesManagerObject.HasValidObjectRef(), CHIP_ERROR_INCORRECT_STATE,
+                        ChipLogError(Zcl, "Invalid mMessagesManagerObject"));
+    VerifyOrReturnError(mPresentMessagesMethod != nullptr, CHIP_ERROR_INCORRECT_STATE,
+                        ChipLogError(Zcl, "mPresentMessagesMethod null"));
+
+    env->ExceptionClear();
+    {
+        char hex_buf[(kMessageIdLength * 2) + 1];
+        VerifyOrReturnError(
+            CHIP_NO_ERROR ==
+                chip::Encoding::BytesToUppercaseHexString(messageId.data(), messageId.size(), hex_buf, sizeof(hex_buf)),
+            CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "BytesToUppercaseHexString failed"));
+
+        UtfString jIdentifier(env, hex_buf);
```

Comment:
it seems UtfString is redundant, where UtfString create globalRef, and release it when it is out of scope in this function, it is better to use mEnv->NewStringUTF(hex_buf). when the data is passed from jni to java, Java would take care of the lifetime and decide when to release it.

Reply:
addressed